File: Java/022.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Arrays;

public class Problem022 {

    public static int nameScore(String name) {
        int score = 0;
        for (char c : name.toCharArray()) {
            score += c - 'A' + 1;
        }
        return score;
    }

    public static void main(String[] args) {
        try {
            String content = new String(Files.readAllBytes(Paths.get("names.txt")));
            String[] names = content.replace("\"", "").split(",");
            Arrays.sort(names);

            long totalScore = 0;
            for (int i = 0; i < names.length; i++) {
                totalScore += (long) (i + 1) * nameScore(names[i]);
            }

            System.out.println(totalScore);

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Rust/022.rs
use std::fs;

fn name_score(name: &str) -> u32 {
    name.chars().map(|c| (c as u32) - ('A' as u32) + 1).sum()
}

fn main() {
    let content = fs::read_to_string("names.txt").expect("Unable to read file");
    let mut names: Vec<String> = content
        .trim()
        .replace('"', "")
        .split(',')
        .map(|s| s.to_string())
        .collect();

    names.sort();

    let mut total_score: u64 = 0;
    for (i, name) in names.iter().enumerate() {
        total_score += ((i + 1) as u64) * (name_score(name) as u64);
    }

    println!("{}", total_score);
}
